runtime/vm/LayoutFFITypeHelpers.cpp

```diff
@@ -156,9 +219,9 @@ LayoutFFITypeHelpers::getStructFFITypeElements(char **layout)
 			nElements += 1;
 			break;
 		}
-		case ']': // End of struct
+		case ']': /* End of struct. */
 			*layout = currentLayout;
```

Comment:
Shouldn't this leave `*layout` pointing at the first character of the next (nested) type?
```c
			*layout = currentLayout + 1;
```

Reply:
No really as we'd like `*layout` to point to the count of element in the struct layout string to be extracted from `LayoutFFITypeHelpers::getStructFFITypeElements` before we parse the elements in the layout string (See https://github.com/eclipse-openj9/openj9/blob/d075ca5a59fb9b6ad646e4e7bdbe999e78c4d1b8/runtime/vm/LayoutFFITypeHelpers.cpp#L135)

Reply:
Line 229 will leave it pointing at the closing `]` of that layout.
Do we need a separate pointer for this loop?
Or is `*layout` irrelevant, in which case we can simplify the function so it operates on a string (`const char *`) instead a pointer to one?

Reply:
> Or is *layout irrelevant, in which case we can simplify the function so it operates on a string (const char *) instead a pointer to one?

Not really. `const char *` doesn't work in our case in that `*layout` must be reassigned to point to the next element (primitive or nested struct) of the outer struct after reaching the boundary of the current struct element when traversing the whole outer struct layout string.